common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java

```diff
@@ -0,0 +1,158 @@
+                Logger.warn(methodTag, "ignoring null broadcast type ");
+            } else {
+                switch (broadcastType) {
+                    case SharedDeviceModeConstants.BROADCAST_TYPE_SDM_REGISTRATION_START:
+                        sharedDeviceModeCallback.onSharedDeviceModeRegistrationStarted();
+                        break;
+                    case SharedDeviceModeConstants.BROADCAST_TYPE_SDM_REGISTERED:
+                        if (isDeviceInSharedMode(context, platformComponents)) {
+                            Logger.info(methodTag, "Device is registered in SDM, clearing default account cache.");
+                            final IAccountCredentialCache accountCredentialCache = new SharedPreferencesAccountCredentialCache(
+                                    new CacheKeyValueDelegate(),
+                                    platformComponents.getStorageSupplier().getEncryptedNameValueStore(
+                                            DEFAULT_ACCOUNT_CREDENTIAL_SHARED_PREFERENCES, String.class)
+                            );
+                            accountCredentialCache.clearAll();
+                            sharedDeviceModeCallback.onSharedDeviceModeRegistered();
+                        } else {
+                            Logger.warn(methodTag, "Device not in shared device mode, ignore broadcast.");
+                        }
+                        break;
+                    case SharedDeviceModeConstants.BROADCAST_TYPE_GLOBAL_SIGN_OUT:
+                        sharedDeviceModeCallback.onGlobalSignOut();
+                        break;
+                    default:
+                        Logger.warn(methodTag, "ignoring unknown broadcast type " + broadcastType);
+                        break;
+                }
+            }
+        } catch (final BaseException e) {
+            Logger.error(methodTag, "Failed to handle SDM broadcast", e);
+        }
+    }
+
+    private static boolean isDeviceInSharedMode(@NonNull final Context context,
+                                                @NonNull IPlatformComponents platformComponents) throws BaseException {
+        final BrokerData activeBroker = BrokerDiscoveryClientFactory.getInstanceForBrokerSdk(context, platformComponents)
+                .getActiveBroker(false);
+        if (activeBroker == null) {
+            return  false;
+        }
+        final BrokerMsalController brokerMsalController = new BrokerMsalController(context, platformComponents, activeBroker.getPackageName());
+        final CommandParameters commandParameters;
+        commandParameters = CommandParameters.builder()
+                .platformComponents(platformComponents)
+                .correlationId(UUID.randomUUID().toString())
+                .build();
+        return brokerMsalController.getDeviceMode(commandParameters);
+    }
+
+    /**
+     * Callback for SDM broadcasts
+     */
+    public interface SharedDeviceModeCallback {
```

Comment:
i'm wondering if we should create an interface based on the actions we want app devs to do (as opposed to the steps in SDM)

e.g.

```Java
interface SDMCallBack{
    // change your UX!
    void enableSDM();

    // clear all data from the app.
    void clearData();
}
``

(in this case, clearUserData will also be invoked when we receive BROADCAST_TYPE_SDM_REGISTERED. but that would be handled in handleSharedDeviceModeBroadCast(). User doesn't need to be aware of it)

Reply:
For now, keeping this as-is to map the callbacks with individual sdm events